### DISCUSSION TITLE
DBeaver IO Error: Cannot open file "*\mydb.duckdb": The process cannot access the file because it is being used by another process.

### DISCUSSION THREAD

**Discussion by Starlord22** [duckdb/duckdb#7054] [General]
Hi I have code snippet to start with below. I will state my approach. Feel free to critique, take any unnecessary steps away from the below.

My approach
1. Create a persistent Duckdb connection here (database='mydb.duckdb')
2. Read a csv file using python.
3. Write csv file into Duckdb table
4. Do some processing, this will come later.
5. I am trying to read my persistent database='mydb.duckdb' from DBeaver.
6. When trying to connect to duckdb from DBeaver, I get 'IO Error: Cannot open file' error

Question: Is there a more optimal way of doing this. i.e. Write data into Duckdb table. Connect to DBeaver. Read table from DBeaver?

What changes do I need to make to my below code snippet to release the connection to the database. So that I can connect to the database from DBeaver?

 
```


import duckdb
import pandas as pd
con = duckdb.connect(database='mydb.duckdb')
c = con.cursor()
df = pd.read_csv(r'\Users\Home\Downloads\duckdb-sql-tools-main\duckdb-sql-tools-main\data\chinook\csv\invoices.csv')
con.register('invoices',df)
result = con.execute("SELECT * FROM invoices")
df1 = pd.DataFrame(result.fetchall())
df1.info(verbose=True)


    <class 'pandas.core.frame.DataFrame'>
    RangeIndex: 412 entries, 0 to 411
    Data columns (total 9 columns):
     #   Column  Non-Null Count  Dtype  
    ---  ------  --------------  -----  
     0   0       412 non-null    int64  
     1   1       412 non-null    int64  
     2   2       412 non-null    object 
     3   3       412 non-null    object 
     4   4       412 non-null    object 
     5   5       210 non-null    object 
     6   6       412 non-null    object 
     7   7       384 non-null    object 
     8   8       412 non-null    float64
    dtypes: float64(1), int64(2), object(6)
    memory usage: 29.1+ KB

```

**Comment by Starlord22** [comment]
Would be great to see examples of implementation of how people are utilizing duckdb to read csv files. Then connect to tools such as DBeaver or any other Database GUI/IDE tools. I am not married to DBeaver. I want a db client that works with Duckdb.

**Reply by Tishj** [reply]
We don't support having multiple processes opening a database in read/write mode, if you open the database in read-only mode in both processes that should work however

**Reply by Starlord22** [reply]
Thanks few questions: 

I am now able to .close() the connection from python code.

However when I attempt to read the database='mydb.duckdb' from DBeaver. I cannot seem to be able to see the table schema? How can I make sure the table I created has persisted?
